ext/php8/handlers_curl.c

```diff
@@ -1,7 +1,12 @@
 #include <php.h>
 #include <stdbool.h>
 
+#if defined(HAVE_CURL) && !defined(COMPILE_DL_CURL)
 #include <ext/curl/php_curl.h>
+#else
+__attribute__((weak)) zend_class_entry *curl_ce = NULL;
+__attribute__((weak)) zend_class_entry *curl_multi_ce = NULL;
+#endif
```

Comment:
Any reason we shouldn't just always do the weak symbols thing? Since we have to use it some of the time, why not all of the time?

Reply:
To summarize our off-thread conversation: Since this is the first time we're introducing weak symbol resolution with `dlsym()` to the project, I'm not confident that we won't inadvertently introduce a regression on the happy path (when the curl header is present) if we go all in on weak symbols.

This technique should solve the issue for this specific edge case while also providing information about how weak symbol resolution with `dlsym()` fits within our existing infrastructure. We might go all in on this method in the future after our testing ability for this edge case matures and we bring more integrations to the extension level, but for now I would feel safer from a stability perspective to keep week symbols as a fallback to the edge case.